d3ploy/demand_driven_deployment_inst.py

```diff
@@ -277,18 +310,23 @@ def calc_diff(self, commod, time):
         return diff, supply, demand
 
     def predict_supply(self, commod):
+        if self.installed_cap:
+            input = self.installed_capacity[commod]
```

Comment:
Input is a built-in function in python.
You can create a variable called input, but you shouldn't. 
Creating such a variable over-rides the input() function, so it's bad practice. 
https://stackoverflow.com/a/20670757/761664